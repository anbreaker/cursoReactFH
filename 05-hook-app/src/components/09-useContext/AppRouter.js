import React from 'react';
import {BrowserRouter as Router, Switch, Route, Redirect} from 'react-router-dom';
import {AboutPage} from './AboutPage';
import {LoginPage} from './LoginPage';
import {HomePage} from './HomePage';
import {Navbar} from './Navbar';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <div className="container">
          <Switch>
            <Route exact path="/about" component={AboutPage} />
            <Route exact path="/login" component={LoginPage} />
            <Route exact path="/" component={HomePage} />
            <Redirect to="/" />
          </Switch>
        </div>
      </div>
    </Router>
  );
};
