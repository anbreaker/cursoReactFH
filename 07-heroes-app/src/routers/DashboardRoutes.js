import React from 'react';
import {Navbar} from '../components/ui/Navbar';
import {Redirect, Route, Switch} from 'react-router-dom';

import {HeroesPage} from '../components/heroes/HeroesPage';
import {MarvelPage} from '../components/marvel/MarvelPage';
import {DcPage} from '../components/dc/DcPage';

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-2">
        <Switch>
          <Route exact path="/marvel" component={MarvelPage} />
          <Route exact path="/hero/:heroeId" component={HeroesPage} />
          <Route exact path="/dc" component={DcPage} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
