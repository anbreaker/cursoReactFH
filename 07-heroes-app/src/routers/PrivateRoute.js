import React from 'react';
import {Redirect, Route} from 'react-router-dom';
import PropTypes from 'prop-types';

export const PrivateRoute = ({isAuthenticate, component: Component, ...rest}) => {
  return (
    <Route
      {...rest}
      component={(props) =>
        isAuthenticate ? <Component {...props} /> : <Redirect to="/login" />
      }
    />
  );
};

PrivateRoute.propTypes = {
  isAuthenticate: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};
