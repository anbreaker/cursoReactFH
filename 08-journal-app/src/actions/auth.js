import {firebase, googleAuthProvider} from '../firebase/firebaseConfig';
import {types} from '../types/types';

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({user}) => {
        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        console.error('Error ->', error);
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({user}) => {
        await user.updateProfile({displayName: name});

        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        console.error('Error ->', error);
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({user}) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};
