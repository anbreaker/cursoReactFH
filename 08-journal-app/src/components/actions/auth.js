import {firebase, googleAuthProvider} from '../../firebase/firebaseConfig';
import {types} from '../../types/types';

export const startEmailPassword = (email, password) => {
  return (dispatch) => {
    setTimeout(() => {
      dispatch(login(123, 'curro'));
    }, 3000);
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({user}) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};
