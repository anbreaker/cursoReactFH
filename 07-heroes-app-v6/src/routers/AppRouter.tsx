import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { LoginPage } from '../components/login/LoginPage';
import { PrivateRoute } from './PrivateRoute';
import { DashboardRoutes } from './DashboardRoutes';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        {/* <Route path="/login" element={<LoginPage />} /> */}

        <Route
          path="/login"
          element={
            <PublicRoute>
              <LoginPage />
            </PublicRoute>
          }
        />

        <Route
          path="/*"
          element={
            <PrivateRoute>
              <DashboardRoutes />
            </PrivateRoute>
          }
        />

        {/* <Route path="/*" element={<DashboardRoutes />} /> */}
      </Routes>
    </BrowserRouter>
  );
};
