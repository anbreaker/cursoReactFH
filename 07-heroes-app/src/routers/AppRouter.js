import React, {useContext} from 'react';

import {BrowserRouter as Router, Switch, Route} from 'react-router-dom';
import {AuthContext} from '../auth/AuthContext';
import {LoginPage} from '../components/login/LoginPage';
import {DashboardRoutes} from './DashboardRoutes';
import {PrivateRoute} from './PrivateRoute';

export const AppRouter = () => {
  const {user} = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/login" component={LoginPage} />

          <PrivateRoute
            path="/"
            component={DashboardRoutes}
            isAuthenticate={user.logged}
          />
        </Switch>
      </div>
    </Router>
  );
};
