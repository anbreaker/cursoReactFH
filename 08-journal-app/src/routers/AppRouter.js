import React, {useEffect, useState} from 'react';
import {BrowserRouter as Router, Switch, Redirect} from 'react-router-dom';
import {useDispatch} from 'react-redux';

import {firebase} from '../firebase/firebaseConfig';
import {AuthRouter} from './AuthRouter';
import {PrivateRoute} from './PrivateRoute';
import {PublicRoute} from './PublicRoute';

import {JournalScreen} from '../components/journal/JournalScreen';
import {login} from '../actions/auth';

export const AppRouter = () => {
  const dispatch = useDispatch();

  const [checking, setChecking] = useState(true);

  const [isLoggedIn, setIsLoggedIn] = useState(false);

  useEffect(() => {
    firebase.auth().onAuthStateChanged((user) => {
      if (user?.uid) {
        dispatch(login(user.uid, user.displayName));
        setIsLoggedIn(true);
        console.log(user.uid, user.displayName, 'esto de aqui...');
      } else {
        setIsLoggedIn(false);
      }
      setChecking(false);
    });
  }, [dispatch, setChecking, setIsLoggedIn]);

  if (checking) {
    return <h1>Loading...</h1>;
  }

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute path="/auth" component={AuthRouter} isAuthenticate={isLoggedIn} />
          {/* <Route exact path="/" component={JournalScreen} /> */}
          <PrivateRoute
            exact
            path="/"
            component={JournalScreen}
            isAuthenticate={isLoggedIn}
          />

          <Redirect to="/auth/login" />
        </Switch>
      </div>
    </Router>
  );
};
