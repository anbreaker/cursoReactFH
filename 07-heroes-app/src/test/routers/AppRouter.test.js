import {mount} from 'enzyme';
import {AuthContext} from '../../auth/AuthContext';
import {AppRouter} from '../../routers/AppRouter';

describe('Pruebas en <AppRouter /> ', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  test('debe mostrar el login si no esta autenticado', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el componente Marvel si esta autenticado', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: 'anbreaker',
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find('.navbar').exists()).toBe(true);
  });
});
