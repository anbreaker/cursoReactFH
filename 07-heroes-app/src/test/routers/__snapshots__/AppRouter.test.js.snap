// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter />  debe mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container"
  >
    <h1>
      LoginPage
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
